src/coreComponents/mesh/CellElementRegionSelector.cpp

```diff
@@ -0,0 +1,259 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2020 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2020 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2020 TotalEnergies
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All rights reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+#include "mesh/CellElementRegionSelector.hpp"
+
+#include <fnmatch.h>
+#include <optional>
+
+
+namespace geos
+{
+using namespace dataRepository;
+using ViewKeys = CellElementRegion::viewKeyStruct;
+
+
+std::optional< string > getCellBlockAttributeValue( string_view cellBlockName )
```

Comment:
separator can be a parameter of this function and a member of CellElementRegionSelector class